diff --git a/src/Makefile b/src/Makefile
index 21cbb2e..4457de2 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -94,6 +94,10 @@ endif
 
 ROOT	?= $(DESTDIR)
 
+ifneq ($(LCRYPT),)
+  SULOGINLIBS	+= $(LCRYPT)
+else
+
 # Additional libs for GNU libc.
 ifneq ($(wildcard $(ROOT)/usr/lib*/libcrypt.*),)
   SULOGINLIBS	+= -lcrypt
@@ -106,6 +110,8 @@ ifneq ($(findstring -lcrypt, $(SULOGINLIBS)), -lcrypt)
 endif
 endif
 
+endif
+
 all:		$(BIN) $(SBIN) $(USRBIN)
 
 #%: %.o
