diff --git a/man/dummy-man b/man/dummy-man
index e298295..73846b7 100755
--- a/man/dummy-man
+++ b/man/dummy-man
@@ -37,17 +37,10 @@ while test $# -gt 0; do
   shift
 done
 
-test $# -gt 0 || fatal_ "missing argument"
-test $# -le 1 || fatal_ "too many non-option arguments"
-
 dist_man=$(printf '%s\n' "$include" | sed 's/\.x$/.1/')
 test -f "$dist_man" && cp "$dist_man" "$output" && exit || :
 
 baseout=`basename_ "$output"`
-sed 's/^/WARNING: /' >&2 <<END
-Did not generate or find default '$baseout' man page.
-Creating a stub man page instead.
-END
 
 progname=`basename_ "$1"`
 bs='\'
