diff --git a/builds/unix/freetype-config.in b/builds/unix/freetype-config.in
index 2d5b90d..1c14aab 100644
--- a/builds/unix/freetype-config.in
+++ b/builds/unix/freetype-config.in
@@ -35,11 +35,11 @@ if test $? -eq 0 ; then
   dynamic_libs=`%PKG_CONFIG% --libs freetype2`
   static_libs=`%PKG_CONFIG% --static --libs freetype2`
 else
-  prefix="%prefix%"
-  exec_prefix="%exec_prefix%"
+prefix="%prefix%"
+  exec_prefix="${prefix}"
 
-  includedir="%includedir%"
-  libdir="%libdir%"
+  includedir="${prefix}/include"
+  libdir="${exec_prefix}/lib"
 
   version=%ft_version%
 
