--- a/Makefile.org	2014-12-06 09:32:55.969612292 +0100
+++ b/Makefile	2014-12-06 11:05:57.432208910 +0100
@@ -10,13 +10,13 @@
 LUA_IMPL         ?= lua
 CC               ?= $(MAC_ENV) gcc
 
-LUA_VERSION       = $(shell $(PKG_CONFIG) --print-provides --silence-errors $(LUA_IMPL))
+#LUA_VERSION       = $(shell $(PKG_CONFIG) --print-provides --silence-errors $(LUA_IMPL))
 ifeq ($(LUA_VERSION),)
 LUA_CMOD         ?= /usr/lib/lua/5.1
 LUA_LMOD         ?= /usr/share/lua/5.1
 LIBDIR           ?= /usr/lib
 LUA_INC          ?= /usr/include
-CURL_LIBS         = -L/usr/lib -lcurl
+CURL_LIBS        ?= -L/usr/lib -lcurl
 else
 LUA_CMOD         ?= $(shell $(PKG_CONFIG) --variable INSTALL_CMOD $(LUA_IMPL))
 LUA_LMOD         ?= $(shell $(PKG_CONFIG) --variable INSTALL_LMOD $(LUA_IMPL))
