--- a/src/subshell/common.c
+++ b/src/subshell/common.c
@@ -1179,7 +1179,7 @@
         "  else "
         "    [ \"${PWD##$HOME/}\" = \"$PWD\" ] && MC_PWD=\"$PWD\" || MC_PWD=\"~/${PWD##$HOME/}\"; "
         "  fi; "
-        "  echo \"${MC_PS1_SAVED:-$USER@$(hostname -s):$MC_PWD\\$ }\"; "
+        "  echo \"${MC_PS1_SAVED:-$USER@$(hostname):$MC_PWD\\$ }\"; "
         "  pwd>&%d; "
         "  kill -STOP $$; "
         "}; "

