diff --git a/Makefile b/Makefile
index 87966a9..cb38ce6 100644
--- a/Makefile
+++ b/Makefile
@@ -90,7 +90,7 @@ endif
 all: $(LIBGIFSO) libgif.a $(LIBUTILSO) libutil.a $(UTILS)
 ifeq ($(UNAME), Darwin)
 else
-	$(MAKE) -C doc
+	#$(MAKE) -C doc
 endif
 
 $(UTILS):: libgif.a libutil.a
@@ -122,7 +122,7 @@ clean:
 	rm -fr doc/*.1 *.html doc/staging
 
 check: all
-	$(MAKE) -C tests
+	#$(MAKE) -C tests
 
 reflow:
 	@clang-format --style="{IndentWidth: 8, UseTab: ForIndentation}" -i $$(find . -name "*.[ch]")
@@ -181,7 +181,7 @@ EXTRAS =     README \
 
 DSOURCES = Makefile *.[ch]
 DOCS = doc/*.xml doc/*.1 doc/*.7 doc/*.html doc/index.html.in doc/00README doc/Makefile
-ALL =  $(DSOURCES) $(DOCS) tests pic $(EXTRAS)
+ALL =  $(DSOURCES) $(EXTRAS)
 giflib-$(VERSION).tar.gz: $(ALL)
 	$(TAR) --transform='s:^:giflib-$(VERSION)/:' -czf giflib-$(VERSION).tar.gz $(ALL)
 giflib-$(VERSION).tar.bz2: $(ALL)
