--- gst-fluendo-mpegdemux-0.10.69/src/gstmpegdefs.h
+++ gst-fluendo-mpegdemux-0.10.69/src/gstmpegdefs.h
@@ -156,6 +156,7 @@
 #define ST_BD_AUDIO_AC3_TRUE_HD         0x83
 #define ST_BD_AUDIO_AC3_PLUS            0x84
 #define ST_BD_AUDIO_DTS_HD              0x85
+#define ST_BD_AUDIO_DTS_HD_MASTER_AUDIO 0x86
 #define ST_BD_AUDIO_EAC3                0x87
 #define ST_BD_PGS_SUBPICTURE            0x90
 #define ST_BD_IGS                       0x91
--- gst-fluendo-mpegdemux-0.10.69/src/gstmpegtsdemux.c
+++ gst-fluendo-mpegdemux-0.10.69/src/gstmpegtsdemux.c
@@ -850,6 +850,8 @@
       break;
     case ST_BD_AUDIO_DTS:
     case ST_PS_AUDIO_DTS:
+    case ST_BD_AUDIO_DTS_HD:
+    case ST_BD_AUDIO_DTS_HD_MASTER_AUDIO:
       template = klass->audio_template;
       name = g_strdup_printf ("audio_%04x", stream->PID);
       caps = gst_caps_new_simple ("audio/x-dts", NULL);
