--- a/ext/rtmp/gstrtmpsrc.c
+++ b/ext/rtmp/gstrtmpsrc.c
@@ -88,7 +88,6 @@
     GValue * value, GParamSpec * pspec);
 static void gst_rtmp_src_finalize (GObject * object);
 
-static gboolean gst_rtmp_src_unlock (GstBaseSrc * src);
 static gboolean gst_rtmp_src_stop (GstBaseSrc * src);
 static gboolean gst_rtmp_src_start (GstBaseSrc * src);
 static gboolean gst_rtmp_src_is_seekable (GstBaseSrc * src);
@@ -144,7 +143,6 @@
 
   gstbasesrc_class->start = GST_DEBUG_FUNCPTR (gst_rtmp_src_start);
   gstbasesrc_class->stop = GST_DEBUG_FUNCPTR (gst_rtmp_src_stop);
-  gstbasesrc_class->unlock = GST_DEBUG_FUNCPTR (gst_rtmp_src_unlock);
   gstbasesrc_class->is_seekable = GST_DEBUG_FUNCPTR (gst_rtmp_src_is_seekable);
   gstbasesrc_class->prepare_seek_segment =
       GST_DEBUG_FUNCPTR (gst_rtmp_src_prepare_seek_segment);
@@ -625,23 +623,6 @@
 #undef STR2AVAL
 
 static gboolean
-gst_rtmp_src_unlock (GstBaseSrc * basesrc)
-{
-  GstRTMPSrc *rtmpsrc = GST_RTMP_SRC (basesrc);
-
-  GST_DEBUG_OBJECT (rtmpsrc, "unlock");
-
-  /* This closes the socket, which means that any pending socket calls
-   * error out. */
-  if (rtmpsrc->rtmp) {
-    RTMP_Close (rtmpsrc->rtmp);
-  }
-
-  return TRUE;
-}
-
-
-static gboolean
 gst_rtmp_src_stop (GstBaseSrc * basesrc)
 {
   GstRTMPSrc *src;
@@ -649,6 +630,7 @@
   src = GST_RTMP_SRC (basesrc);
 
   if (src->rtmp) {
+    RTMP_Close (src->rtmp);
     RTMP_Free (src->rtmp);
     src->rtmp = NULL;
   }

