diff --git a/src/ui/xupnpd_ui.lua b/src/ui/xupnpd_ui.lua
index 9236780..d19398b 100644
--- a/src/ui/xupnpd_ui.lua
+++ b/src/ui/xupnpd_ui.lua
@@ -14,7 +14,7 @@ end
 
 function ui_error()
     http.send('<h2>Error occurred</h3>')
-    http.send('<br/><a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_downloads()
@@ -26,7 +26,7 @@ function ui_downloads()
         end
     end
     http.send('</table>')
-    http.send('<br/><a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_download(name)
@@ -87,7 +87,7 @@ function ui_playlists()
     http.send('<input type=file name=m3ufile><br /><br />')
     http.send('<input class="btn btn-primary" type=submit value=Send>')
     http.send('</form><hr/>')
-    http.send('<br/><a class="btn btn-primary" href="/ui/reload">Reload</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/playlists">Reload feeds</a> <a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-primary" href="/ui/reload">Reload</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/playlists">Reload feeds</a> <a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_feeds()
@@ -121,7 +121,7 @@ function ui_feeds()
     http.send('<br/><input class="btn btn-primary" type=submit value=Add> <a class="btn btn-info" href="/ui/fhelp" target="_blank">Help</a>')
     http.send('</form><hr/>')
 
-    http.send('<br/><a class="btn btn-primary" href="/ui/save_feeds">Save</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/feeds">Reload feeds</a> <a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-primary" href="/ui/save_feeds">Save</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/feeds">Reload feeds</a> <a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_fhelp()
@@ -429,7 +429,7 @@ function ui_status()
 
     http.send('</table>')
 
-    http.send('<br/><a class="btn btn-primary" href="/ui/status">Refresh</a> <a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-primary" href="/ui/status">Refresh</a> <a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_kill()
