diff --git a/src/ui/ui_config.html b/src/ui/ui_config.html
index 784f27e..afce5af 100644
--- a/src/ui/ui_config.html
+++ b/src/ui/ui_config.html
@@ -204,5 +204,5 @@
 <a class='btn btn-primary' href='/ui/restart'>Restart</a>
 <a class='btn btn-info' href='/ui/plugins'>Plugins</a>
 <a class='btn btn-info' href='/ui/profiles'>Profiles</a>
-<a class='btn btn-info' href='/ui'>Back</a>
+<a class='btn btn-info' href='/ui/'>Back</a>
 </form>
diff --git a/src/ui/xupnpd_ui.lua b/src/ui/xupnpd_ui.lua
index dbd9c7d..863c5b3 100644
--- a/src/ui/xupnpd_ui.lua
+++ b/src/ui/xupnpd_ui.lua
@@ -11,7 +11,7 @@ end
 
 function ui_error()
     http.send('<h2>Error occurred</h3>')
-    http.send('<br/><a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_downloads()
@@ -23,7 +23,7 @@ function ui_downloads()
         end
     end
     http.send('</table>')
-    http.send('<br/><a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_download(name)
@@ -84,7 +84,7 @@ function ui_playlists()
     http.send('<input type=file name=m3ufile><br /><br />')
     http.send('<input class="btn btn-primary" type=submit value=Send>')
     http.send('</form><hr/>')
-    http.send('<br/><a class="btn btn-primary" href="/ui/reload">Reload</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/playlists">Reload feeds</a> <a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-primary" href="/ui/reload">Reload</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/playlists">Reload feeds</a> <a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_feeds()
@@ -118,7 +118,7 @@ function ui_feeds()
     http.send('<br/><input class="btn btn-primary" type=submit value=Add> <a class="btn btn-info" href="/ui/fhelp" target="_blank">Help</a>')
     http.send('</form><hr/>')
 
-    http.send('<br/><a class="btn btn-primary" href="/ui/save_feeds">Save</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/feeds">Reload feeds</a> <a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-primary" href="/ui/save_feeds">Save</a> <a class="btn btn-primary" href="/ui/reload_feeds?return_url=/ui/feeds">Reload feeds</a> <a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_fhelp()
@@ -425,7 +425,7 @@ function ui_status()
 
     http.send('</table>')
 
-    http.send('<br/><a class="btn btn-primary" href="/ui/status">Refresh</a> <a class="btn btn-info" href="/ui">Back</a>')
+    http.send('<br/><a class="btn btn-primary" href="/ui/status">Refresh</a> <a class="btn btn-info" href="/ui/">Back</a>')
 end
 
 function ui_kill()
