diff --git a/libavformat/avio.c b/libavformat/avio.c
index 4846bbd8c6..978bf72994 100644
--- a/libavformat/avio.c
+++ b/libavformat/avio.c
@@ -216,12 +216,12 @@ int ffurl_connect(URLContext *uc, AVDictionary **options)
                (uc->protocol_whitelist && !strcmp(uc->protocol_whitelist, e->value)));
     av_assert0(!(e=av_dict_get(*options, "protocol_blacklist", NULL, 0)) ||
                (uc->protocol_blacklist && !strcmp(uc->protocol_blacklist, e->value)));
-
+/*
     if (uc->protocol_whitelist && av_match_list(uc->prot->name, uc->protocol_whitelist, ',') <= 0) {
         av_log(uc, AV_LOG_ERROR, "Protocol '%s' not on whitelist '%s'!\n", uc->prot->name, uc->protocol_whitelist);
         return AVERROR(EINVAL);
     }
-
+*/
     if (uc->protocol_blacklist && av_match_list(uc->prot->name, uc->protocol_blacklist, ',') > 0) {
         av_log(uc, AV_LOG_ERROR, "Protocol '%s' on blacklist '%s'!\n", uc->prot->name, uc->protocol_blacklist);
         return AVERROR(EINVAL);
