diff --git a/scripts/kconfig/confdata.c b/scripts/kconfig/confdata.c
index 8f4ecbd..b73fe02 100644
--- a/scripts/kconfig/confdata.c
+++ b/scripts/kconfig/confdata.c
@@ -395,7 +395,7 @@ int conf_write(const char *name)
 		use_timestamp = 0;
 
 	fprintf(out, _("#\n"
-		       "# Automatically generated make config: don't edit\n"
+		       "# Automatically generated make config: do not edit\n"
 		       "# Busybox version: %s\n"
 		       "%s%s"
 		       "#\n"),
