diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7f03c35..2be3cbc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.15)
+cmake_minimum_required(VERSION 3.7)
 
 include(CheckIncludeFileCXX)
 include(CheckIPOSupported)
@@ -6,7 +6,7 @@ include(CheckIPOSupported)
 project(ninja)
 
 # --- optional link-time optimization
-check_ipo_supported(RESULT lto_supported OUTPUT error)
+#check_ipo_supported(RESULT lto_supported OUTPUT error)
 
 if(lto_supported)
 	message(STATUS "IPO / LTO enabled")
