diff --git a/src/Makefile b/src/Makefile
index 9ae35df..814b0bd 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -10,7 +10,7 @@
 
 CPPFLAGS =
 CFLAGS  ?= -O2
-override CFLAGS += -ansi -fomit-frame-pointer -fstack-protector-strong -W -Wall -Wunreachable-code -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -D_XOPEN_SOURCE -D_GNU_SOURCE -DVERSION=\"$(VERSION)\"
+override CFLAGS += -ansi -fomit-frame-pointer -W -Wall -Wunreachable-code -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -D_XOPEN_SOURCE -D_GNU_SOURCE -DVERSION=\"$(VERSION)\"
 override CFLAGS += $(shell getconf LFS_CFLAGS)
 STATIC	=
 MANDB	:= s@^\('\\\\\"\)[^\*-]*-\*- coding: [^[:blank:]]\+ -\*-@\1@
