--- a/util-linux/mount.c	2014-02-24 21:48:18.000000000 +0100
+++ b/util-linux/mount.c	2014-02-24 21:49:20.000000000 +0100
@@ -2083,6 +2083,14 @@
 
 	// Open either fstab or mtab
 	fstabname = "/etc/fstab";
+	//NI use /var/etc/fstab if accessible
+	FILE * f= fopen ("/var/etc/fstab", "r");
+	if (f) {
+		fstabname = "/var/etc/fstab";
+		fclose(f);
+	}	
+	printf("mount: using %s\n", fstabname);
+
 	if (i & MS_REMOUNT) {
 		// WARNING. I am not sure this matches util-linux's
 		// behavior. It's possible util-linux does not
