diff --git a/Makefile b/Makefile
index b2a62cf..c199335 100644
--- a/Makefile
+++ b/Makefile
@@ -108,8 +108,16 @@ echo:
 pc:
 	@echo "version=$R"
 	@echo "prefix=$(INSTALL_TOP)"
+	@echo "exec_prefix=$(INSTALL_TOP)"
 	@echo "libdir=$(INSTALL_LIB)"
 	@echo "includedir=$(INSTALL_INC)"
+	@echo ""
+	@echo "Name: Lua"
+	@echo "Description: An Extensible Extension Language"
+	@echo "Version: $R"
+	@echo "Requires:"
+	@echo "Libs: -L\$${libdir} -llua -lm -ldl"
+	@echo "Cflags: -I\$${includedir}"
 
 # list targets that do not create files (but not all makes understand .PHONY)
 .PHONY: all $(PLATS) clean test install local none dummy echo pecho lecho
